ui/app/AppLayouts/Wallet/panels/SwapInputPanel.qml

```diff
@@ -121,6 +121,17 @@ Control {
         }
     }
 
+    /* TODO: remove after https://github.com/status-im/status-desktop/issues/15604 is
+    implemented as this is hack to set token values correctly when model is reset */
+    Connections {
+        target: holdingSelector.model
+        function onRowsInserted() {
```

Comment:
does this catch all potential changes in the holdings model?

Reply:
yep cause the adaptor uses concat model which emits rowsInserted and on modelReset and currently thats the nly scenario thats valid when refreshing assets on nim side. Which should change under https://github.com/status-im/status-desktop/issues/15604